### PR TITLE
Insert data to contents table

### PR DESCRIPTION
**Pull request by zaher-aa**
- Insert data to contents table and added new field to it `primary_category`

### CODE REVIEW

**Review by AymanSami98 — APPROVED** [review]

**Review by RandSohail — APPROVED** [review]

File: src/interfaces/index.ts [hunk 1]
```diff
@@ -24,14 +24,31 @@ interface IListUserWatched {
...
   runtime: number;
-  type: string;
 }
 
+interface IVLContent extends IContent {
+  primaryCategory: { title: string };
+}
+interface ISysContent extends IContent {
+  primaryCategory: string ;
+}
```
Comment (Jood80) [inline src/interfaces/index.ts:51]: Just wondering, in your `ISysContent` do you need all of `IContent` attributes?

If Yes, ignore what I'm about to write. 
If No, and you want to assure the compatibility between these two interfaces, we can see that example:
```ts
interface Location {
  address: string;
  x: number;
  y: number;
}

interface SpecificLocation extends Omit<Location, 'address' | 'x' | 'y'> {
  address: {
    country: string;
    city: string;
    street: string;
  };
  x: string;
  y: string;
}
``` 
[The Resource](https://bobbyhadz.com/blog/typescript-override-interface-property)

File: src/interfaces/index.ts [hunk 2]
```diff
@@ -60,16 +77,45 @@ interface IListUserWatchedReturn {
...
+  updatedBy: number;
+  tvodTicketsCount: number;
+  tvodSeconds: number;
+  title?: string;
+}
+
+interface IPayment {
+  contentId: string;
+  monetizationModel: 'tvod' | 'svod';
+  type: 'PAYMENT' | 'REFUND';
```
Comment (Jood80) [inline src/interfaces/index.ts:95]: The `type` is capitalized, whereas `monetizationModel` is not.
Is that how you receive it from viewlift API?

File: src/utilities/viewlift/requests/listPurchases.ts [hunk 3]
```diff
@@ -50,8 +58,11 @@ const getAllPurchases = async (purchasesData: any): Promise<any> => {
       }
 
       return { allPurchases };
-    } catch (err: any) {
-      throw new CustomError(err.message, 'error while fetching purchases');
+    } catch (err) {
+      if (err instanceof Error) {
+        throw new CustomError(err.message, 'error while fetching purchases');
```
Comment (Jood80) [inline src/utilities/viewlift/requests/listPurchases.ts:63]: What's the difference between these two messages. Why one can't be enough to describe the crisis?

File: src/database/models/Content.ts [hunk 4]
```diff
@@ -6,22 +6,23 @@ import sequelize from '../connections';
 interface IContent extends Model<
 InferAttributes<IContent>, InferCreationAttributes<IContent>
 > {
-  id?: number;
+  id: string;
   userId?: number;
   title: string;
-  viewliftPublishDate: string;
+  publishDate: number;
```
[the remark this answers is not in the document]
Comment (zaher-aa) [reply src/database/models/Content.ts:12]: We need to stick with viewlift api namings, `publishDate` is what they use

File: src/interfaces/index.ts (hunk 1 above)
Comment (zaher-aa) [reply src/interfaces/index.ts:51]: No, we need it all `IContent`

File: src/interfaces/index.ts (hunk 2 above)
Comment (zaher-aa) [reply src/interfaces/index.ts:95]: Yes, this is what we received from viewlift API

File: src/utilities/viewlift/requests/listPurchases.ts (hunk 3 above)
Comment (zaher-aa) [reply src/utilities/viewlift/requests/listPurchases.ts:63]: the first message is the regular error message. 
the second message is for making it easy for us when debugging the code an know where the error happened.

**Review by Ibrahim-Jarada — APPROVED** [review]

**Review by Jood80 — APPROVED** [review]

**Pull request merged**